### PR TITLE
[OFFAPPS-1120] Rely on API parameter, not our own logic

### PR DESCRIPTION
**Pull request by ocke**
## Description
Removed `is the user editable` logic and rely on the API call.
Fixed UI by using zendesk garden for error messages.

Go by commit for easy reviewing.

## References
[JIRA ticket](https://zendesk.atlassian.net/browse/OFFAPPS-1120)

## CCs
@zendesk/apps-migration
@zendesk/vegemite 

## Risks
* [medium] user and/or organisation details/notes can't be edited.

### CODE REVIEW

**Comment by sostopher** [comment]
I won't review the actual code but yay at this PR

File: package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
   "dependencies": {
```
Comment (sostopher) [inline package.json]: Can we bump package version as well as manifest?

File: src/javascript/app.js [hunk 2]
```diff
@@ -32,13 +32,13 @@ const app = {
...
     storage('organizationFields', null)
     storage('userFields', null)
-    storage('userEditable', true)
 
     app.getInformation().then(() => {
       app.fillEmptyStatuses(storage('ticketsCounters'))
       app.fillEmptyStatuses(storage('orgTicketsCounters'))
       app.showDisplay()
     }).catch((err) => {
+      console.error(err)
```
[the remark this answers is not in the document]
Comment (sostopher) [reply src/javascript/app.js]: Yes. This is in response to a few tickets we got today about an error being thrown in the app but nothing logged in the console or elsewhere. 

#4415758, #4411937 if you want to look.

File: src/javascript/app.js [hunk 3]
```diff
@@ -78,9 +78,9 @@ const app = {
       const getUserFieldsPromise = ajax('getUserFields')
       promises.push(getUserFieldsPromise)
 
-      // We need to make sure getCustomRolesPromise is done, because it sets 'userEditable'.
-      // getCustomRolesPromise can be undefined, but that's not a problem for Promise.all
-      Promise.all([getUserFieldsPromise, getCustomRolesPromise]).then(([userFieldsData]) => {
+      // We need to make sure all promiss are done, because they set certain storage values.
+      Promise.all(promises).then((data) => {
+        const userFieldsData = data.pop()
```
[the remark this answers is not in the document]
Comment (ocke) [reply src/javascript/app.js]: I reordered the promises for this to be clearer

**Pull request merged**